embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java

```diff
@@ -82,12 +90,361 @@ public PageOutput open(TaskSource taskSource, Schema inputSchema,
         String getRule();
     }
 
+    private interface CharacterTypesRule
+            extends Rule {
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("replace")
+        @ConfigDefault("\"_\"")
+        String getReplace();
+    }
+
+    private interface FirstCharacterTypesRule
+            extends Rule {
+        @Config("replace")
+        @ConfigDefault("null")
+        Optional<String> getReplace();
+
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("prefix")
+        @ConfigDefault("null")
+        Optional<String> getPrefix();
+    }
+
+    private interface TruncateRule
+            extends Rule {
+        @Config("max_length")
+        @ConfigDefault("128")
+        int getMaxLength();
+    }
+
+    private interface RegexReplaceRule
+            extends Rule {
+        @Config("match")
+        String getMatch();
+
+        @Config("replace")
+        String getReplace();
+    }
+
+    private interface UniqueNumberSuffixRule
+            extends Rule {
+        @Config("delimiter")
+        @ConfigDefault("\"_\"")
+        String getDelimiter();
+
+        @Config("digits")
+        @ConfigDefault("null")
+        Optional<Integer> getDigits();
+
+        @Config("esteem_original_names")
+        @ConfigDefault("true")
+        boolean getEsteemOriginalNames();
+
+        @Config("max_length")
+        @ConfigDefault("null")
+        Optional<Integer> getMaxLength();
+
+        @Config("offset")
+        @ConfigDefault("1")
```

Comment:
It can be 0 (but non-negative). I'd add a documentation later that the `offset` sets the index of the first duplication.

offset == 0: id, id_1, id_2, id_3
offset == 1: id, id_2, id_3, id_4

I set 1 as its default since the offset == 1 case looks more natural.